### PR TITLE
hpos-admin: implement schema validation

### PR DESCRIPTION
**Pull request by yegortimoshenko**

### CODE REVIEW

File: overlays/holo-nixpkgs/hpos-admin/hpos-admin.py [hunk 1]
```diff
@@ -60,7 +61,12 @@ def put_settings():
             app.logger.warning('CAS mismatch: {} != {}'.format(received_cas, expected_cas))
             return '', 409
         state['v1']['settings'] = request.get_json(force=True)
-        replace_file_contents(get_state_path(), json.dumps(state, indent=2))
+        state_json = json.dumps(state, indent=2)
+        try:
+            subprocess.run(['hpos-config-is-valid'], check=True, input=state_json, text=True)
+        except CalledProcessError:
```
Comment (samrose) [inline overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]: Currently this line is throwing `machine# [   20.153227] hpos-admin[646]: NameError: name 'CalledProcessError' is not defined `

fwiw

Comment (samrose) [inline overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]:
```suggestion
        except CalledProcessError:
```
```suggestion
        except subprocess.CalledProcessError:
```

Comment (peeech) [reply overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]: Yeah this will fix the error you're describing

**Comment by peeech** [comment]
Hydra also is reporting the error:
```
FileNotFoundError: [Errno 2] No such file or directory: 'hpos-config-is-valid': 'hpos-config-is-valid'
```
which means `hpos-config-is-valid` needs to be added to PATH of `hpos-admin.service`. I suggest adding to [systemd.services.hpos-admin](https://github.com/Holo-Host/holo-nixpkgs/blob/2020-02-27-hpos-admin-schema/modules/services/hpos-admin.nix#L33) the line:
```
path = with pkgs; [ hpos-config-is-valid ];
```

**Comment by yegortimoshenko** [comment]
@peeech @samrose Addressed your feedback.

**Pull request merged**